### ISSUE TITLE
Cannot install package on Ubuntu 19.10 and earlier due to libgcc-s1

### ISSUE THREAD

**Issue by rufusshrestha** [dandavison/delta#504]
Have we dropped support for Ubuntu 16.04?

Cannot install due to libgcc-s1 dependency error

`Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies.
 git-delta : Depends: libgcc-s1 (>= 4.2) but it is not installable
E: Unable to correct problems, you have held broken packages.
`

**Comment by mnsgs** [comment]
I am seeing the same on Ubuntu 19.10 on 0.5.0. Prior versions worked fine

**Comment by Kr1ss-XD** [comment]
@mnsgs 19.10 has reached end-of-life in July 2020, hence I would suggest upgrading to 20.04-LTS or 20.10.

As for 16.04 @rufusshrestha - Looks like `libgcc-s1` is not in the `xenial` repositories; the earliest release with that package that I could find is `focal` 20.04. Maybe it's about time to upgrade the system ?  General maintainance for 16.04 will end in a few weeks, anyways.

**Comment by filipelbc** [comment]
Same issue here. But on Ubuntu 18.04

**Comment by rufusshrestha** [comment]
@Kr1ss-XD Thanks Kris. If it is no longer supported for 16.04 then its fine with EOL coming up on Apr anyway. However, I still think its a regression if it does not work on 18.04 reported by @filipelbc and use of library only used in 20.04 will invalidate an LTS release that I think should be supported. As my issue is resolved, I'm not sure if I should leave this open until 18.04 issue is resolved or close it.

**Comment by Kr1ss-XD** [comment]
> I still think its a regression if it does not work on 18.04

I definitely agree about that. I'd suggest to leave the issue open. @dandavison mentioned he'd be off for two weeks, but maybe there's still some way to support Ubuntu's before 20.4 (probably this also hits older Debian versions as well).

Maybe you could change the issue title, so that it says 'Ubuntu <20.4' or '19.10 and earlier' or something like that ?

**Comment by rufusshrestha** [comment]
@Kr1ss-XD Thank you. Sure, will do as advised. No hurry there @dandavison I'm sure we can all live with older version for a while.

**Comment by Kr1ss-XD** [comment]
**NB/**	Looks like this actually affects Debian 9 (oldstable) and 10 (stable) as well : 

The `libgcc-s1` seems to be [available](https://packages.debian.org/search?suite=all&keywords=libgcc-s1) only for Debian 11 `Bullseye` (testing) and `Sid` (unstable).

**Comment by rufusshrestha** [comment]
On Ubuntu 16.04, th

**Comment by rufusshrestha** [comment]
On Ubuntu 16.04, this can be installed by manually fetching the following packages from focal repo as a temp workaround

gcc-10-base_10.2.1-6_amd64.deb
libgcc-s1_10.2.1-6_amd64.deb

**Comment by dandavison** [comment]
Thanks all. Do you have a recommended action for delta to take here?

**Comment by ritzmann** [comment]
The version linked against musl is working fine on Debian Buster. Maybe you could simply sit this problem out and refer x64 users to the musl package?

**Comment by filipelbc-ba** [comment]
The version linked against musl works fine in Ubuntu 18.04.

**Comment by jamesob** [comment]
This is basically just an informational post, no recommended course of action, but I thought I might post some related links that may be helpful.

I've been affected by this as well. Running on Debian 10 (stable) with apt configuration to pull in from testing (bullseye) when necessary (but not preferentially). When installing with `dpkg -i git-delta_0.6.0_amd64.deb`, apt seems to autoremove `cryptsetup-initramfs`. Subsequently, I see the dreaded 
```
Volume group "debian-vg" not found
Cannot process volume group debian-vg
Gave up waiting for suspend/resume device
```
on boot. In order to attempt a fix, I had to boot from a live CD, chroot (per [this process](https://gist.github.com/samuelcolvin/43c5ed2807e7db004b1058d0c9bfb068)), and `apt install cryptsetup cryptsetup-initramfs`. I then ran `update-grub`, but my boot's still trashed.

Here's a link to a similar (if not identical) issue with `bat`: https://github.com/sharkdp/bat/issues/1371
Here's a link to perhaps the underlying Debian bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=950551

Wish I had a useful recommendation, but people who are mixing apt sources need to beware.

(Hi @dandavison! Awesome project you've got here.)

**Comment by dandavison** [comment]
Argh! Hello @jamesob and I sincerely hope that is the worst outcome that anyone will ever have as a result of using delta.

I've added a warning and suggestion to use the musl-linked versions to the README.  If anyone can suggest anything more we can do then please do but I'll close this as it sounds like there's not a lot.

**Comment by jli** [comment]
**tl;dr** if you're on an older version of Ubuntu (19.10 and earlier) or Debian (10 and earlier), go to the [delta releases](https://github.com/dandavison/delta/releases) page, search for `git-delta-musl`, download that, and install with `sudo dpkg -i <path-to-deb-file>`

I also ran into a conflict with an existing `delta` package. I uninstalled that, and then installing the git-delta .deb worked.

**Comment by dandavison** [comment]
Would anyone here be able to review the small change to the .deb being made in #1217? (Removes "provides")
